recommendation/microsoft.owin.json

```diff
@@ -136,17 +136,60 @@
               "TargetCPU": [ "x86", "x64", "ARM32", "ARM64" ]
             }
           ],
-          "Description": "Add a comment to explain how to remove the base class of OwinMiddleware.",
+          "Description": "Remove the base class for owin middleware, remove the override method modifier for the invoke method, add a new gloable xpression for RequestDelegate _next, initialize it in the constructor and remove the constructor intializer base(next).",
           "Actions": [
             {
-              "Name": "AddComment",
+              "Name": "RemoveBaseClass",
+              "Type": "Class",
+              "Value": "OwinMiddleware",
+              "Description": "Remove the base class OwinMiddleware",
+              "ActionValidation": {
+                "Contains": "",
+                "NotContains": ":OwinMiddleware"
+              }
+            },
+            {
+              "Name": "ReplaceMethodModifiers",
               "Type": "Class",
-              "Value": "Please remove the base class here : OwinMiddleware",
-              "Description": "Add a comment to remove the base class OwinMiddleware",
+              "Value": {
+                "methodName": "Invoke",
+                "modifiers": "public"
+              },
+              "Description": "Replace all the invoke method modifiers with only public in order to remove the override modifier.",
               "ActionValidation": {
-                "Contains": "Pleaseremovethebaseclasshere:OwinMiddleware",
+                "Contains": "public",
+                "NotContains": "override"
+              }
+            },
+            {
+              "Name": "AddGlobalExpression",
+              "Type": "Class",
+              "Value": "RequestDelegate _next = null;",
+              "Description": "Add a new global variable to replace the OwinMiddleware global abstract variable of Next.",
+              "ActionValidation": {
+                "Contains": "RequestDelegate_next=null;",
```

Comment:
RequestDelegate_next=null;, I know we have discussed this.. please find some better way to represent this... a) After replace add a commented line RequestDelegate  _next = null; // CTA Changes and validate now (RequestDelegate  _next = null; // CTA Changes) , b)Keep the Actionvalidation as separate config , c)Change ActionValidation to TestValidation (NOOP).. so that people just ignore this section...  Create an issue to address this, and continue with check in

Reply:
Thanks Biroj, Good feedback I will go with option A for now and revisit this in the future.